internal/pkg/service/common/etcdop/op/noresult.go

```diff
@@ -29,7 +29,7 @@ func NewNoResultOp(factory Factory, mapper NoResultMapper) NoResultOp {
 	return out
 }
 
-func (v NoResultOp) Do(ctx context.Context, client *etcd.Client) (err error) {
+func (v NoResultOp) Do(ctx context.Context, client etcd.KV) (err error) {
```

Comment:
Tu nam staci interface, cast klienta, nemusi tu byt cela struktura.
Etcd libka, totiz umoznuje tieto interfaces rozne `wrappovat` a tak menit spravania sa klienta, pomocou kompozicii.

Nakoniec som to nevyuzil, ale interfaces som tu nechal.